tests/beta/interfaces/test_tape_tf.py

```diff
@@ -785,3 +783,571 @@ def test_sampling(self):
+
+        from pennylane import numpy as anp
+
+        a = anp.array(0.1, requires_grad=True)
+
+        res1 = circuit(a)
+        grad_fn = qml.grad(circuit)
+        grad1 = grad_fn(a)
+
+        # switch to TF interface
+        circuit.to_tf()
+
+        a = tf.Variable(0.1, dtype=tf.float64)
+
+        with tf.GradientTape() as tape:
+            res2 = circuit(a)
+
+        grad2 = tape.gradient(res2, a)
+        assert np.allclose(res1, res2, atol=tol, rtol=0)
+        assert np.allclose(grad1, grad2, atol=tol, rtol=0)
+
+    def test_jacobian(self, dev_name, diff_method, mocker, tol):
+        """Test jacobian calculation"""
+        spy = mocker.spy(QuantumTape, "jacobian")
+        a = tf.Variable(0.1, dtype=tf.float64)
+        b = tf.Variable(0.2, dtype=tf.float64)
+
+        dev = qml.device(dev_name, wires=2)
+
+        @qnode(dev, diff_method=diff_method, interface="tf")
+        def circuit(a, b):
+            qml.RY(a, wires=0)
+            qml.RX(b, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return [expval(qml.PauliZ(0)), expval(qml.PauliY(1))]
+
+        with tf.GradientTape() as tape:
+            res = circuit(a, b)
+
+        assert circuit.qtape.trainable_params == {0, 1}
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2,)
+
+        expected = [tf.cos(a), -tf.cos(a) * tf.sin(b)]
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        res = tape.jacobian(res, [a, b])
+        expected = [[-tf.sin(a), tf.sin(a) * tf.sin(b)], [0, -tf.cos(a) * tf.cos(b)]]
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        if diff_method == "finite-diff":
+            spy.assert_called()
+        elif diff_method == "backprop":
+            spy.assert_not_called()
+
+    def test_jacobian_dtype(self, dev_name, diff_method, tol):
+        """Test calculating the jacobian with a different datatype"""
+        if diff_method == "backprop":
+            pytest.skip("Test does not support backprop")
+
+        a = tf.Variable(0.1, dtype=tf.float32)
+        b = tf.Variable(0.2, dtype=tf.float32)
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qnode(dev)
+        def circuit(a, b):
+            qml.RY(a, wires=0)
+            qml.RX(b, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return [expval(qml.PauliZ(0)), expval(qml.PauliY(1))]
+
+        circuit.to_tf(dtype=tf.float32)
+        assert circuit.dtype is tf.float32
+
+        with tf.GradientTape() as tape:
+            res = circuit(a, b)
+
+        assert circuit.qtape.interface == "tf"
+        assert circuit.qtape.trainable_params == {0, 1}
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2,)
+        assert res.dtype is tf.float32
+
+        res = tape.jacobian(res, [a, b])
+        assert [r.dtype is tf.float32 for r in res]
+
+    def test_jacobian_options(self, dev_name, diff_method, mocker, tol):
+        """Test setting jacobian options"""

```

Comment:
(Minor suggestion:) Maybe adding here that these would be specific to finite diff?